File: loadercomplex-forge/src/ver122/java/com/enderzombi102/loadercomplex/forge12/impl/item/ForgeItemStack.java
package com.enderzombi102.loadercomplex.forge12.impl.item;

import com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack;
import org.jetbrains.annotations.NotNull;

public class ForgeItemStack implements ItemStack {
	private final net.minecraft.item.ItemStack stack;

	public ForgeItemStack( net.minecraft.item.ItemStack stack ) {
		this.stack = stack;
	}

	public @NotNull Object getStack() {
		return this.stack;
	}

	@Override
	public String getDisplayName() {
		return this.stack.getDisplayName();
	}

	@Override
	public void setDisplayName( String newName ) {
		this.stack.setStackDisplayName( newName );
	}

	@Override
	public @NotNull Object getType() {
		return this.stack.getItem();
	}

	@Override
	public int getAmount() {
		return this.stack.getCount();
	}

	@Override
	public void setAmount( int amount ) {
		this.stack.setCount( amount );
	}

	@Override
	public void setDamage( int damage ) {
		this.stack.setItemDamage( damage );
	}

	@Override
	public int getDamage() {
		return this.stack.getItemDamage();
	}

	@Override
	public int getMaxDurability() {
		return this.stack.getMaxDamage();
	}

	@Override
	public int getMaxStackSize() {
		return this.stack.getMaxStackSize();
	}
}

File: loadercomplex-forge/src/ver122/kotlin/com/enderzombi102/loadercomplex/forge122/impl/item/ForgeItemStack.kt
package com.enderzombi102.loadercomplex.forge122.impl.item

import net.minecraft.item.ItemStack

class ForgeItemStack(private val stack: ItemStack) : com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack {
	override fun getStack(): Any {
		return this.stack
	}

	override fun getDisplayName(): String? {
		return stack.displayName
	}

	override fun setDisplayName(newName: String) {
		stack.setStackDisplayName(newName)
	}

	override fun getType(): Any {
		return stack.item
	}

	override fun getAmount(): Int {
		return stack.count
	}

	override fun setAmount(amount: Int) {
		stack.count = amount
	}

	override fun setDamage(damage: Int) {
		stack.itemDamage = damage
	}

	override fun getDamage(): Int {
		return stack.itemDamage
	}

	override fun getMaxDurability(): Int {
		return stack.maxDamage
	}

	override fun getMaxStackSize(): Int {
		return stack.maxStackSize
	}
}
